db/migrate/20231021124102_replace_users_permissions_by_join_table.rb

```diff
@@ -0,0 +1,7 @@
+class ReplaceUsersPermissionsByJoinTable < ActiveRecord::Migration[7.1]
+  def change
+    drop_table :users_permissions
+
+    create_join_table :permissions, :users
+  end
+end
```

Comment:
oh ok why the join table rather than the userPermissions?

Reply:
This creates the join table between users and permissions, which will be named `permission_user`, and enables the use of `has_many_and_belongs_to_many` in both models. Does that answer your question?

Reply:
oh so keeping the users_permissions would have kept the has_many_and_belongs_to_many but only in the direction of user -> permissions?

Reply:
No, the other solution was a bad one from me, done by hand without knowing this was the correct way. This assures the correct relationship of many-to-many in the database. The `user_permissions` table allowed for a one-to-many solution only.